core/controller/src/main/scala/whisk/core/controller/WebActions.scala

```diff
@@ -483,7 +483,20 @@ trait WhiskWebActionsApi extends Directives with ValidateRequestSize with PostAc
               provide(fullyQualifiedActionName(actionName)) { fullActionName =>
                 onComplete(verifyWebAction(fullActionName, onBehalfOf.isDefined)) {
                   case Success((actionOwnerIdentity, action)) =>
-                    if (!action.annotations.getAs[Boolean]("web-custom-options").exists(identity)) {
+                    val requireWebAuthIsBool = (action.annotations.getAs[Boolean](WhiskAction.requireWhiskAuthAnnotation) != None)
+                    val annotationRequireWebAuth = action.annotations
+                      .getAs[Int](WhiskAction.requireWhiskAuthAnnotation).map(_.toString)
+                      .orElse(action.annotations.getAs[String](WhiskAction.requireWhiskAuthAnnotation))
+                    val enforceWhiskAuthHdr = (!requireWebAuthIsBool && annotationRequireWebAuth.isDefined)
```

Comment:
Please rename to `enforceWhiskAuthHdr`, took me a full 10 seconds to figure out it's supposed to mean "header" vs. High Dynamic Range.